lib/feature/setting/data/repositories/setting_repository_impl.dart

```diff
@@ -0,0 +1,25 @@
+import 'package:job_interview_practice/core/services/local_data_store.dart';
+import 'package:job_interview_practice/feature/setting/data/models/setting.dart';
+import 'package:job_interview_practice/feature/setting/data/repositories/setting_repository.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+class SettingRepositoryImpl implements SettingRepository {
+  // appの起動時に呼ばれる想定。全ての設定情報を持ったインスタンスをサービスで作成するため、
+  // ここではdatastoreからsettingmodelを撮ってくる
+  List<SettingModel> selectAll() {
+    LocalDataStore localStore = LocalDataStore()..connectToStore();
+    localStore.prefs.get('questionNumber');
+  }
+
+  SettingModel selectBykey(String key) {
+    LocalDataStore localStore = LocalDataStore()..connectToStore();
+    String value = localStore.prefs.getString(key);
+    return SettingModel(value: value, name: key);
+  }
+
+  void updateSetting(String key, String value) async {
+    LocalDataStore localStore = LocalDataStore()..connectToStore();
+    localStore.prefs = await SharedPreferences.getInstance();
```

Comment:
Around here, I initialized LocalDataStore in all times  to use want use `localStore.prefs`.
I want to use `connectToStore()`below in `lib/core/services/local_data_store.dart`.
Is there good solution ??

```
import 'package:shared_preferences/shared_preferences.dart';

class LocalDataStore {
  SharedPreferences prefs;

  void connectToStore() async {
    this.prefs = await SharedPreferences.getInstance();
  }
}
```

Reply:
nope, the whole of this PR needs refactor, let's meet and discuss it

Reply:
The main problem is with LocalDataStore, it should be divided into two sections:
1. DataStoreProvider -> service class which has a CRUD methods -> create/read/update/delete and makes everything on SharedPreferences
2. LocalDataStore -> class which will be depending on `feature` but will be independent of 3rd party libraries like SharedPreferences, so it will do only one action which is direct connected with whole `feature`